CommunityProject/Assets/_Game/Scripts/Editor/OnBuildDestroyObjsManager.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace BoundfoxStudios.CommunityProject.Editor
+{
+	public class OnBuildDestroyObjsManager : IProcessSceneWithReport
+	{
+		public int callbackOrder { get { return 0; } }
+		public void OnProcessScene(Scene scene, BuildReport report)
+		{
+			if(report == null)return;
+			if (!report.summary.options.HasFlag(UnityEditor.BuildOptions.Development))
+			{
+				GameObject[] objToDestroyOnBuild = GameObject.FindGameObjectsWithTag("DevelopmentOnly");
+				int objToDestroyOnBuildLength = objToDestroyOnBuild.Length;
+				for (int i = 0; i < objToDestroyOnBuildLength; i++)
+				{
+					GameObject.DestroyImmediate(objToDestroyOnBuild[i], true);
+				}
+			}
+
```

Comment:
```suggestion
      if (report is null || report.summary.options.HasFlag(BuildOptions.Development)) {
        return;
      }

			var objectsToDelete = GameObject.FindGameObjectsWithTag("DevelopmentOnly");

			for (var i = objectsToDelete.Length - 1; i >= 0; i--)
			{
			  Object.DestroyImmediate(objectsToDelete[i]);
			}
```

Hab ein paar Dinge hier drin korrigiert:

* Wir bevorzugen `var` anstelle den Typen im Community-Projekt.
* Zuerst prüfen, ob wir im Development-Modus sind und dann direkt die Methode beenden. Spart an Einrückungstiefe
* Beim Zerstören von Elementen sollte man immer rückwärts durch die Schleife laufen. Es kann durchaus mal passieren, dass Dinge ineinander verschaltelt sind, dann würde ein Eltern-Objekt auch die Kinder löschen und es kommt dann zum Fehler, da wir versuchen in der nächsten Iteration auf die Kinder zuzugreifen.